src/size.rs

```diff
@@ -329,6 +329,30 @@ impl<T: PartialEq, U> TypedSize2D<T, U> {
     }
 }
 
+impl<T: Float, U> TypedSize2D<T, U> {
```

Comment:
why is this `Float` instead of, say, `PartialOrd`?

Reply:
The answer to your question is, looking at the git history, [a commit in 2015](https://github.com/servo/euclid/commit/84150a904899957cb32e1c54fa31ad9100937507). I suppose what you really meant was "It'd be nice to use the PartialOrd trait bound instead" and I agree, so I'm going ahead and making that change.

Reply:
D'oh! actually euclid is `#[no_std]` which means no `PartialOrd`, and `num_traits` only provides min/max functions for `Float`s. The only way I can think of to get min and max for integers is add an std feature flag and provide two implementations for min/max/clamp. But since that's quite a bit more complicated I'd rather wait for someone to complain about the functions missing.